frontend/src/pages/ProfilePage.tsx

```diff
@@ -0,0 +1,13 @@
+import UserProfile from "../components/ProfileComponent.tsx";
+import {useParams} from "react-router-dom";
+
+export default function ProfilePage() {
+
+    const {username} = useParams<{ username: string }>()
```

Comment:
```suggestion
    const {username} = useParams<{ username?: string }>();  // in case undefined
```

Or alternatively:
`<UserProfile username={username ?? ""} />`